### PR TITLE
Icu 7490 UI add new address field to target

### PR DESCRIPTION
**Pull request by ZedLi**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-7490)

## Description
Added the new address field to target. The user will be prompted to delete host sources if they exist when adding an address and vice versa.

### Screenshots (if appropriate):
image

image

image

### CODE REVIEW

File: ui/admin/app/routes/scopes/scope/targets.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ export default class ScopesScopeTargetsRoute extends Route {
    */
   @action
   @loading
-  @notifyError(({ message }) => message)
+  @notifyError(({ message }) => message, { catch: true })
```
Comment (ZedLi) [inline ui/admin/app/routes/scopes/scope/targets.js:58]: I catch here so as to not display any errors twice as the form is now also calling this save and also using `notifyError`.

**Review by cameronperera — APPROVED** [review]
LGTM!

**Review by calcaide — APPROVED** [review]
LGTM! Awesome job!

While testing this I found what I think is a bug for an edge case, I will DM you steps I follow to reproduce and a screen capture of the process.

Not sure if is a bug of the new implementation or we are missing something in Mirage that is making it look like a bug.

**Comment by ZedLi** [comment]
> While testing this I found what I think is a bug for an edge case, I will DM you steps I follow to reproduce and a screen capture of the process.

Great catch! The issue was after removing the hosts, the model gets updated from the response which blows away the address field so I had to set it again right before saving.

**Review by calcaide — APPROVED** [review]
AMAZING WORK!!! Tested again and I could not reproduce the previous bug 😄

**Pull request merged**